README.md

```diff
@@ -1,142 +1,56 @@
-add signals, parameters and other behaviours. See the modulo component API documentation for more information.
-
-For the AICA application framework to discover the components, they must be registered.
-
-To register the class as a component, the following macro is needed at the end of the source file,
-where `template_component_package` is replaced by the local package name and `CPPComponent` is replaced by the component name:
-
-```cpp
-#include "rclcpp_components/register_node_macro.hpp"
-
-RCLCPP_COMPONENTS_REGISTER_NODE(template_component_package::CPPComponent)
-```
-
-Then, add the following lines to CMakeLists.txt to generate and register the component library target.
-The target name `cpp_component` can be replaced with any unique identifier for the particular component.
-
-```cmake
-ament_auto_add_library(cpp_component SHARED ${PROJECT_SOURCE_DIR}/src/CPPComponent.cpp)
-rclcpp_components_register_nodes(cpp_component "${PROJECT_NAME}::CPPComponent")
-list(APPEND COMPONENTS cpp_component)
+```bash
+./rename_package.sh my_custom_package [--dry-run]
 ```
 
-## Writing Python components
+If you want to include a second ROS package within the AICA package, you will need to create a new package folder in `source`
+and add it in your `aica-package.toml` file under `[build.packages.name_of_new_package]` accordingly.
 
-Similar to C++, a Python component is a class that inherits from either `modulo_components.component.Component` or
-`modulo_components.lifecycle_component.LifecycleComponent`.
+## Configure the package development environment
 
-Refer to the modulo component API documentation for more information.
+This template uses a devContainer configuration ([`.devcontainer.json`](./.devcontainer.json)) with base AICA Docker images
+for a seamless integrated development experience.
 
-To register the Python class as a component, add an entrypoint to setup.cfg under the `python_components` field.
-Replace `template_component_package` and `PyComponent` with the package and component name respectively.
+Using VSCode and the Dev Containers extension, after creating and renaming the template package, simply open the repository
+in a devcontainer using the "Reopen in Container" command.
 
-```conf
-python_components =
-    template_component_package::PyComponent = template_component_package.py_component:PyComponent
+Other IDEs such as JetBrains can similarly be configured to use development containers.
 
-```
+If any changes are made to aica-package.toml (including any package names or dependencies), remember to rebuild the devcontainer.
 
 ## Building your component package
 
-You will need to have Docker installed in order to build your component package (see [Docker installation](https://docs.docker.com/get-docker/)).
-
-Then you can build your component using the following command:
+You can build your package using the following command:
 
 ```bash
 docker build -f aica-package.toml .
```

Comment:
No need for an image name?

Reply:
Well technically no if you just want to build or test it. You get the image SHA and can tag it as normal. Two sentences later we have:

> As we are using `docker build` to build you can pass any Docker argument, like `-t <image_name>` to tag the image 

There is also the later section:

```
## Using your component package

After you have built and tagged your package as a docker image, [...]
```

Reply:
I mean sure it's there but wouldn't hurt to put it anyway no?

Reply:
if that's a blocking comment for you then I can add `-t component-template-package`, but the name doesn't really matter until you want to package it in an app (which is separately documented). By putting it explicitly I kind of feel like people will blindly copy and paste or assume it has to be exactly that tag, without thinking what the tag means or how to use it